GETTING_STARTED.md

```diff
@@ -0,0 +1,374 @@
++store.todos.add({ id: 32, title: 'an other' })
+
++console.log(store.todos.get(2)) // b.
+```
+
+Now you can use the store :
+  - (a) We added a todo, the id is the default key used by keyValue
+  - (b) We retrieve (and log) the todo that has the key equals to `2`
+
+## 3. Provide the store to React (context)
+```diff
++import React from 'react'
+import { createStore, types } from 'k-ramel'
++import { provider } from '@k-ramel/react'
+
+const store = createStore(
+  {
+    todos: types.keyValue(),
+  },
+)
+
+store.todos.add({ id: 2, title: 'finish the documentation' })
+store.todos.add({ id: 32, title: 'an other' })
+
+console.log(store.todos.get(2))
++
++const App = () => (
++  <div>Hey!</div>
++)
++
++export default provider(store)(App)
+```
+
+Now the store is provided in React context, we can use it and bind our a component to it!
+
+## 4. Bind a React component
+```diff
+import React from 'react'
+import { createStore, types } from 'k-ramel'
+-import { provider } from '@k-ramel/react'
++import { provider, inject } from '@k-ramel/react'
+
+const store = createStore(
+  {
+    todos: types.keyValue(),
+  },
+)
+
+store.todos.add({ id: 2, title: 'finish the documentation' })
+store.todos.add({ id: 32, title: 'an other' })
+
+console.log(store.todos.get(2))
++
++ // a.
++const Todos = ({ todos }) => (
++  <ul>
++    {todos.map(({ id, title }) => (
++      <li key={id}>{title}</li>
++    ))}
++  </ul>
++)
++
++ // b.
++const TodosContainer = inject((store) => {
++  return {
++    todos: store.todos.getAsArray(),
++  }
++})(Todos)
+
+const App = () => (
+-  <div>Hey!</div>
++  <TodosContainer /> // c.
+)
+
+export default provider(store)(App)
+```
+
+We create a classical React Component (a) and inject todos (as array) in its props (b), and finally we use this new component in our App (c)
+
+## 5. Add a new todos (store mutation)
+```diff
+import React from 'react'
+import { createStore, types } from 'k-ramel'
+import { provider, inject } from '@k-ramel/react'
+
+const store = createStore(
+  {
+    todos: types.keyValue(),
+  },
+)
+
+store.todos.add({ id: 2, title: 'finish the documentation' })
+store.todos.add({ id: 32, title: 'an other' })
+
+console.log(store.todos.get(2))
+
+-const Todos = ({ todos }) => (
++const Todos = ({ todos, onClick }) => (
++  <div>
++    <button onClick={onClick}>Add a todo!</button> {/* c. */}
+    <ul>
+      {todos.map(({ id, title }) => (
+        <li key={id}>{title}</li>
+      ))}
+    </ul>
++  </div>
+)
+
+const TodosContainer = inject((store) => {
+  return {
+    todos: store.todos.getAsArray(),
++    // a.
++    onClick: () => {
++      // b.
++      store.todos.add({ id: Math.random(), title: 'Yo! I am a new todo!' })
++    }
+  }
+})(Todos)
+
+const App = () => (
+  <TodosContainer />
+)
+
+export default provider(store)(App)
+```
+
+When we click on the new button (c), the function injected (a) is triggered. This function ask for a mutation (b), which add a new todo (the title is hardcoded here)
```

Comment:
Is new really useful ?

the function injected -> the injected function 

triggered -> called